core/domain/classifier_services.py

```diff
@@ -214,6 +214,13 @@ def save_classifier(classifier):
         classifier_id = _create_classifier(classifier)
     else:
         _update_classifier(classifier_data_model, classifier.state_name)
+
+    # Create the mapping from <exp_id,exp_version,state_name>
+    # to classifier_id.
+    create_classifier_exploration_mapping(
```

Comment:
I guess purpose of this method was to update state_name whenever it changed (may be lazily). I don't think we need this method now. Since we either forward old classifier or train a new. If we forward old classifier then it is still valid for old state_name as well as new state_name. So updating state_name is unnecessary (I think we should not update it. This way we can preserve the state_name for which classifier was trained). This means that once classifier is stored it is never going to change.

I think the old classifier model was designed to map them to exploration directly, and we changed that when we introduced mapping table approach. So when we save classifier what it should mean is store the new classifier and map it to exploration.

What do you guys think?